File: src/jobs/handlers/sendSms.js
const { sendSmsViaTwilio } = require('../../providers/messaging');

module.exports = async function handleSendSms(job, context) {
  const { payload } = job;
  if (!payload?.to) {
    throw new Error('SMS payload missing "to"');
  }

  if (!payload.body) {
    throw new Error('SMS payload missing "body"');
  }

  const fromNumber = payload.from || context.tenantConfig.twilioFromNumber;
  if (!fromNumber) {
    throw new Error('SMS payload missing "from" and tenant has no default number');
  }

  await sendSmsViaTwilio({
    tenantConfig: context.tenantConfig,
    to: payload.to,
    body: payload.body,
    from: fromNumber
  });
};

File: src/jobs/handlers/send_sms.py
from src.providers.messaging import send_sms_via_twilio


def handle_send_sms(job, context):
    """Handle send_sms job type."""
    payload = job['payload']

    if not payload.get('to'):
        raise Exception('SMS payload missing "to"')

    if not payload.get('body'):
        raise Exception('SMS payload missing "body"')

    from_number = payload.get('from') or context['tenant_config'].get('twilio_from_number')
    if not from_number:
        raise Exception('SMS payload missing "from" and tenant has no default number')

    send_sms_via_twilio(
        tenant_config=context['tenant_config'],
        to=payload['to'],
        body=payload['body'],
        from_number=from_number
    )
